src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/TypeFixupSignature.cs

```diff
@@ -73,6 +73,17 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
         {
             DependencyList dependencies = new DependencyList();
             dependencies.Add(factory.NecessaryTypeSymbol(_typeDesc), "Type referenced in a fixup signature");
+
+            if (_typeDesc.HasInstantiation && !_typeDesc.IsGenericDefinition)
```

Comment:
Note that this will potentially run for each fixup kind (i.e. if the same type is used with a cast, its static bases are accessed, typeof is used, etc., each of those will run this logic and add all methods on the same type).

It might be cleaner to add a new node type (e.g. `class AllMethodsOnTypeNode : DependencyNodeCore<NodeFactory> { }` that this node depends on and that node depends on the methods.

I don't have a strong opinion though.

If we do the extra intermediary node, it will be easy to extend this in the future to e.g. also include methods on base types (to cover "new genericness" discovered through bases e.g. `class Foo<T> : Bar<T, int> { }` making sure methods on Bar are generated too).

Reply:
I'll give it a try, it sounds like a good idea.

Reply:
Yeah, that should be a lot cleaner, and result in fewer edges in the graph. (Extra edges aren't very expensive, but they aren't free either)